src/software/physics/physics.cpp

```diff
@@ -25,3 +29,133 @@ Vector calculateFutureVelocity(const Vector &initial_velocity,
     double vy1 = initial_velocity.y() + seconds_in_future * constant_acceleration.y();
     return Vector(vx1, vy1);
 }
+
+EuclideanToFourWheel::EuclideanToFourWheel(double front_wheel_angle_deg,
+                                           double back_wheel_angle_deg)
+{
+    auto robot_constants = create2021RobotConstants();
+
+    // TODO: replace with Thunderloop polling rate constant
+    // Set to period of control loop.
+    delta_t_ = 1.0 / 200;
+
+    // import robot constants
+    robot_mass_M_ = robot_constants.mass_kg;
+    // TODO: update
+    robot_radius_R_ = 1;
+    mass_distribution_alpha_ =
+        robot_constants.moment_of_inertia_kg_m_2 / (robot_mass_M_ * robot_radius_R_);
+    // TODO: update
+    front_wheel_angle_phi_ = robot_constants.front_wheel_angle_deg;
+    // TODO: update
+    rear_wheel_angle_theta_ = robot_constants.back_wheel_angle_deg;
+
+    // calculate coupling matrices
+    // ref: http://robocup.mi.fu-berlin.de/buch/omnidrive.pdf
+    auto a = pow(sin(front_wheel_angle_phi_), 2) - pow(cos(front_wheel_angle_phi_), 2);
+    auto b = -sin(front_wheel_angle_phi_) * sin(rear_wheel_angle_theta_) -
+             cos(front_wheel_angle_phi_) * cos(rear_wheel_angle_theta_);
+    auto c = -sin(front_wheel_angle_phi_) * sin(rear_wheel_angle_theta_) +
+             cos(front_wheel_angle_phi_) * cos(rear_wheel_angle_theta_);
+    auto d = pow(sin(rear_wheel_angle_theta_), 2) - pow(cos(rear_wheel_angle_theta_), 2);
+    Eigen::Matrix4d e;
+    e << 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1;
+    Eigen::Matrix4d f;
+    f << 1, a, b, c, a, 1, c, b, b, c, 1, d, c, b, d, 1;
+
+    wheel_force_to_delta_wheel_speed_D_C_alpha_ =
+        1 / (robot_mass_M_ * mass_distribution_alpha_) * e + 1 / robot_mass_M_ * f;
+
+    auto i = 1 / (2 * sin(front_wheel_angle_phi_) + 2 * sin(rear_wheel_angle_theta_));
+    auto j = cos(front_wheel_angle_phi_) / (2 * pow(cos(front_wheel_angle_phi_), 2) +
+                                            2 * pow(cos(front_wheel_angle_phi_), 2));
+    auto k = sin(rear_wheel_angle_theta_) /
+             (2 * sin(front_wheel_angle_phi_), 2 * sin(rear_wheel_angle_theta_));
+
+    wheel_speed_to_euclidean_velocity_D_inverse_ << -i, -i, i, i, j, -j, -(1. - j),
+        (1. - j), k, k, (1 - k), (1 - k);
+}
+
+WheelSpace_t EuclideanToFourWheel::get_target_wheel_speeds(
```

Comment:
we use `camelCase` for our function names and  `snake_case` for our variables

variables look good, but the functions need to be reformatted to `snakeCase` i.e `getTargetWheelSpeeds`